memtable/vectorrep.cc

```diff
@@ -103,16 +107,32 @@ class VectorRep : public MemTableRep {
   using Bucket = std::vector<const char*>;
   std::shared_ptr<Bucket> bucket_;
   mutable port::RWMutex rwlock_;
+  RelaxedAtomic<size_t> bucket_size_;
   bool immutable_;
   bool sorted_;
   const KeyComparator& compare_;
+  // Thread-local vector to buffer concurrent writes.
+  ThreadLocalPtr tl_writes_;
 };
 
 void VectorRep::Insert(KeyHandle handle) {
   auto* key = static_cast<char*>(handle);
-  WriteLock l(&rwlock_);
-  assert(!immutable_);
-  bucket_->push_back(key);
+  {
+    WriteLock l(&rwlock_);
+    assert(!immutable_);
+    bucket_->push_back(key);
+  }
+  bucket_size_.FetchAddRelaxed(1);
+}
+
+void VectorRep::InsertConcurrently(KeyHandle handle) {
+  void* v = tl_writes_.Get();
```

Comment:
It makes more sense to me to put the static_cast before assigning to v from void*, rather than on use of v.